src/components/molecules/InputDate/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react'
+import { updateDepartureDay, updateReturnDay } from 'store/filterSlice'
+import { useAppDispatch } from 'store/hooks'
+import Calendar from '../Calendar'
+import { getPrettyDate } from './functions'
+import './style.scss'
+
+interface IProps {
+  id: string
+  placeholder?: string
+  date?: number | null
+}
+
+export default function InputDate(props: IProps) {
+  const dispatch = useAppDispatch()
+  const [inputValue, setInputValue] = useState('')
+  const [closeCalendar, setCloseCalendar] = useState(false)
+
+  const handleClickInput = () => {
+    setCloseCalendar(prev => !prev)
+  }
+
+  const handleClickCalendar = (date: any) => {
+    if (props.id === 'destination') {
+      dispatch(updateDepartureDay(date.valueOf()))
+    } else {
+      dispatch(updateReturnDay(date.valueOf()))
+    }
+    setInputValue(getPrettyDate(date))
+  }
+
+  return (
+    <>
+      <input
+        onClick={handleClickInput}
+        className="search-form__input"
+        value={inputValue}
+        type="text"
+        placeholder={props.placeholder}
+        readOnly
+      />
+      {closeCalendar && <Calendar onClick={handleClickCalendar} />}
+      <span className="date"></span>
+    </>
```

Comment:
Плохая практика использовать фрагменты, лучше обернуть компонент в контейнер, иначе при верстке возможно неочивидное поведение

```suggestion
    <div>
      <input
        onClick={handleClickInput}
        className="search-form__input"
        value={inputValue}
        type="text"
        placeholder={props.placeholder}
        readOnly
      />
      {closeCalendar && <Calendar onClick={handleClickCalendar} />}
      <span className="date"></span>
    <div/>
```